examples/AvroGeneric/Program.cs

```diff
@@ -80,8 +77,12 @@ static void Main(string[] args)
             CancellationTokenSource cts = new CancellationTokenSource();
             var consumeTask = Task.Run(() =>
             {
-                using (var consumer = new Consumer<string, GenericRecord>(consumerConfig, new AvroDeserializer<string>(), new AvroDeserializer<GenericRecord>()))
+                using (var serdeProvider = new AvroSerdeProvider(avroConfig))
+                using (var consumer = new Consumer<string, GenericRecord>(consumerConfig, serdeProvider.GetDeserializerGenerator<string>(), serdeProvider.GetDeserializerGenerator<GenericRecord>()))
                 {
+                    consumer.OnError += (_, e)
```

Comment:
it will be lost. the plan ultimately is to do some caching on startup and trigger events when the handler is added. this is not ideal, but having iterated over pretty much every other API option, i'm convinced it's the best all things considered. it's the only tolerable API choice imo.

Reply:
Okay, loosing events is not okay though, so this needs to be solved prior to release.
You could queue events until a delegate is set, but if the application never sets one they'll just queue up. Using a timeout, say 60s, is not foolproof either because the application might set up the Kafka client, do something else for a prolonged period of time, and then try to use it.

API delicacy is all good, but not at the expense of correctness.